### ISSUE TITLE
train_pix2pix the result is not good

### ISSUE THREAD

**Issue by 555jion** [GaParmar/img2img-turbo#82]
Dear author，
I greatly admire your research. I am a graduate student from Beijing Forestry University in China, and I would like to apply it to the field of terrain generation in images. But I encountered some problems in implementing the paired training model in this paper, and I hope you can continue to advise me.
Here are my relevant configurations and training process. It seems that the loss function has not converged and the generated results are not satisfactory.
accelerate_config:
        compute_environment: LOCAL_MACHINE
        debug: false
        distributed_type: 'NO'
        downcast_bf16: 'no'
        enable_cpu_affinity: true
        gpu_ids: all
        machine_rank: 0
        main_training_function: main
        mixed_precision: bf16
        num_machines: 1
        num_processes: 1
        rdzv_backend: static
        same_network: true
        tpu_env: []
        tpu_use_cluster: false
        tpu_use_sudo: false
        use_cpu: false

accelerate launch src/train_pix2pix_turbo.py \
      --pretrained_model_name_or_path="stabilityai/sd-turbo" \
      --output_dir="output/pix2pix_turbo/fill50k" \
      --dataset_folder="data/my_fill50k" \
      --resolution=512 \
      --train_batch_size=1 \
      --enable_xformers_memory_efficient_attention --viz_freq 25 \
      --track_val_fid \
      --report_to "w
      --tracker_project_name "pix2pix_turbo_fill50k"
![Screenshot from 2024-08-28 10-58-46](https://github.com/user-attachments/assets/7ff11ca8-1782-48a6-ad33-318e88271d57)

![Screenshot from 2024-08-28 10-58-59](https://github.com/user-attachments/assets/225c03cf-0a93-44ea-a683-4bbb9b52b662)

![Screenshot from 2024-08-28 10-59-11](https://github.com/user-attachments/assets/fb5392c0-4822-4faa-8f83-243357e4dd01)

**Comment by GaParmar** [comment]
Hi, 

I think your training results look worse because of two things: precision="bf16" and batch_size=1. 
Could you try making the following changes to your setting:
-make sure you are using the latest version of the training code. 
-increase the batch_size to 2
-do not use `bf16` mixed precision training

-Gaurav

**Comment by 555jion** [comment]
Thank you for your reply. I will reconfigure and train according to your suggestion. However, I have configured it to always burst the video memory when batch_2 is 2. I would like to inquire about your specific configuration

**Comment by 555jion** [comment]
Thank you for your reply. I will reconfigure and train according to your suggestion, but I have configured it to always burst memory when batch_size set to 2. I would like to inquire about your specific configuration.

**Comment by Lucarqi** [comment]
@555jion hi, I encountered the same issue. The generated quality is very poor. I set BS=2, but in the validation set, each group of images is very similar, and in the test set, all the images are identical. This confuses me a lot. Have you solved this problem?
Training:
![image](https://github.com/user-attachments/assets/7f7a7082-0c2f-479d-9b6a-a4c8058b7bc8)
![image](https://github.com/user-attachments/assets/b841b2b6-9166-4c33-b212-b0e3c7e135d4)
On validation set:
![image](https://github.com/user-attachments/assets/91afa000-ec21-4dae-916e-ceaa3dc8cd4e)
Testing(generation):
![image](https://github.com/user-attachments/assets/e625244e-4c30-4a41-b20a-f0eecf2c5c1a)
![image](https://github.com/user-attachments/assets/f453d2bb-2976-4a36-8f43-d45bada3e73b)
![image](https://github.com/user-attachments/assets/57c6a7a0-0c38-4452-8604-dcec6b588362)
![image](https://github.com/user-attachments/assets/dd3a2f74-989c-428a-b4eb-6089c559f330)

**Comment by 555jion** [comment]
> @555jion hi, I encountered the same issue. The generated quality is very poor. I set BS=2, but in the validation set, each group of images is very similar, and in the test set, all the images are identical. This confuses me a lot. Have you solved this problem? Training: ![image](https://private-user-images.githubusercontent.com/11365027/374456325-7f7a7082-0c2f-479d-9b6a-a4c8058b7bc8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzg5MDUsIm5iZiI6MTczMDAzODYwNSwicGF0aCI6Ii8xMTM2NTAyNy8zNzQ0NTYzMjUtN2Y3YTcwODItMGMyZi00NzlkLTliNmEtYTRjODA1OGI3YmM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE0MTY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmMjIwOTE1MzQzNGM5NDY2MDdkYzk2MmQxMmZhMTMyZTY0NDcyZjljMjgzNzAzNzNkMGFlMzJjNDhiNWY3YmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.s8yTpV-MHsGOQwYR1IQeQiop4j4myLZUKdmM8GohHB0) ![image](https://private-user-images.githubusercontent.com/11365027/374456822-b841b2b6-9166-4c33-b212-b0e3c7e135d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzg5MDUsIm5iZiI6MTczMDAzODYwNSwicGF0aCI6Ii8xMTM2NTAyNy8zNzQ0NTY4MjItYjg0MWIyYjYtOTE2Ni00YzMzLWIyMTItYjBlM2M3ZTEzNWQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE0MTY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3OTc3YjAwMjZiNDI2ZDRjNDhmNzgxM2Q2ZTdmNjY3ZjExMjkxNDdkZjliN2RhY2JlMDMzNWRjZmVkYWMyYWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Zo-ZWJJFoJ5y_pOQcsRKqLytfOqbxokk5PqteQ2hZec) On validation set: ![image](https://private-user-images.githubusercontent.com/11365027/374456665-91afa000-ec21-4dae-916e-ceaa3dc8cd4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzg5MDUsIm5iZiI6MTczMDAzODYwNSwicGF0aCI6Ii8xMTM2NTAyNy8zNzQ0NTY2NjUtOTFhZmEwMDAtZWMyMS00ZGFlLTkxNmUtY2VhYTNkYzhjZDRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE0MTY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1ZDIzZjI4MjA0MzMzOTIyMWNiNDE1YWRkYzlmNDc3ZTkyN2ZmY2UzODMzNzg2YmIwMDE1NzQ1YWRkNTk3YWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.F3kZiyjTPtusIwQMCU6ccIGpyrsmtG4mGg44CC4QO-4) Testing(generation): ![image](https://private-user-images.githubusercontent.com/11365027/374457257-e625244e-4c30-4a41-b20a-f0eecf2c5c1a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzg5MDUsIm5iZiI6MTczMDAzODYwNSwicGF0aCI6Ii8xMTM2NTAyNy8zNzQ0NTcyNTctZTYyNTI0NGUtNGMzMC00YTQxLWIyMGEtZjBlZWNmMmM1YzFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE0MTY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0MDZjNmIzOTkxN2U3YzE5ZjljNDI1MzAwNGE5OWYzOGEyMjZhMDc1N2JmNzFmNjI4YWEwZWI5MjgyMWFmMDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bXu2EHPrWjEB-dJPY0yXTUs9ozrKSpuY1_IwHpw-x40) ![image](https://private-user-images.githubusercontent.com/11365027/374457294-f453d2bb-2976-4a36-8f43-d45bada3e73b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzg5MDUsIm5iZiI6MTczMDAzODYwNSwicGF0aCI6Ii8xMTM2NTAyNy8zNzQ0NTcyOTQtZjQ1M2QyYmItMjk3Ni00YTM2LThmNDMtZDQ1YmFkYTNlNzNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE0MTY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwY2E5MDE2ZDkwOGQwZGNhZDQzNzIxNDFkNGFhNDQ3NjM1NmEwNzEwZTMwZmM3Mjk1M2E5MTI2MmRkMzJiZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8OR3zOtSQpsWBhagk9JzK59Z6RsLqGy6KY4ZVZew6hU) ![image](https://private-user-images.githubusercontent.com/11365027/374457345-57c6a7a0-0c38-4452-8604-dcec6b588362.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzg5MDUsIm5iZiI6MTczMDAzODYwNSwicGF0aCI6Ii8xMTM2NTAyNy8zNzQ0NTczNDUtNTdjNmE3YTAtMGMzOC00NDUyLTg2MDQtZGNlYzZiNTg4MzYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE0MTY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkMjAyYmVmYWQxODgzYWU3MzAxNjg3YmEzMGZkMTU2OTRjMTFmZGY1ZTJmNTAzNmIxMDcxN2UyNjNiYjkzZmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tRYYIzUGMIPL3h9MNZL4eJ9Yf0ZXKwql7peB6GLBIcY) ![image](https://private-user-images.githubusercontent.com/11365027/374457397-dd3a2f74-989c-428a-b4eb-6089c559f330.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAwMzg5MDUsIm5iZiI6MTczMDAzODYwNSwicGF0aCI6Ii8xMTM2NTAyNy8zNzQ0NTczOTctZGQzYTJmNzQtOTg5Yy00MjhhLWI0ZWItNjA4OWM1NTlmMzMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDE0MTY0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkMzE0Zjg3ZTcyZWM3ZjhiMzNhMjU4ZGE4ZDI2YjgwYjM5ZjQxYzVjZWY4OWEwMzVkMDU2M2JhZmRjNzZmMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dNBIKjjE3f3J4uyKCNLWslEwPKnzcLbtzVvhtjScy84)

Hello, it seems that the loss function is converging normally, but it is true that your Fid function is oscillating. Is this related to your text input? I am just guessing that my situation is slightly different from yours.

**Comment by lzh1-hub** [comment]
> 尊敬的作者， 非常钦佩您的研究成果。我是北京林业大学的研究生，希望将其应用到图像中的地形生成领域。但是在实现本文中的配对训练模型时遇到了一些问题，希望您能继续指正。 以下是我的相关配置和训练过程。看起来损失函数没有收敛，生成的结果也不理想。accelerate_config : compute_environment: LOCAL_MACHINE debug: false distributed_type: 'NO' downcast_bf16: 'no' enable_cpu_affinity: true gpu_ids: all machine_rank: 0 main_training_function: main combined_precision: bf16 num_machines: 1 num_processes: 1 rdzv_backend: static same_network: true tpu_env: [] tpu_use_cluster: false tpu_use_sudo: false use_cpu: false
> 
> 加速启动 src/train_pix2pix_turbo.py --pretrained_model_name_or_path="stabilityai/sd-turbo" --output_dir="output/pix2pix_turbo/fill50k" --dataset_folder="data/my_fill50k" --resolution=512 --train_batch_size=1 --enable_xformers_memory_efficient_attention --viz_freq 25 --track_val_fid --report_to "w --tracker_project_name "pix2pix_turbo_fill50k" ![Screenshot from 2024-08-28 10-58-46](https://github.com/user-attachments/assets/7ff11ca8-1782-48a6-ad33-318e88271d57)
> 
> ![Screenshot from 2024-08-28 10-58-59](https://github.com/user-attachments/assets/225c03cf-0a93-44ea-a683-4bbb9b52b662)
> 
> ![Screenshot from 2024-08-28 10-59-11](https://github.com/user-attachments/assets/fb5392c0-4822-4faa-8f83-243357e4dd01)

请问您修改后复现的结果怎么样？我也遇到和您一样的问题，复现的结果并不好
